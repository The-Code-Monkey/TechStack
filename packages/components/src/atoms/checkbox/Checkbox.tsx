import { Interactable, Icon } from '../../primatives';
import { generateAutomationId } from '../../utils';

import { disabledStyles, checkedStyles } from './styled';

export interface Props {
  autoid?: string;
  onClick?: (p1: boolean, p2: MouseEvent) => void;
  checked?: boolean;
  indeterminate?: boolean;
  disabled?: boolean;
}

const Checkbox = ({
  indeterminate,
  checked,
  onClick,
  disabled,
  autoid,
}: Props) => {
  const handleOnClick = (event: MouseEvent) => {
    const status = indeterminate || !checked;

    if (onClick) {
      onClick(status, event);
    }
  };

  return (
    <Interactable
      display='flex'
      size={7}
      alignItems='center'
      justifyContent='center'
      cursor='pointer'
      bgColor='neutrals.0'
      border='1'
      borderRadius='1'
      borderColor='neutrals.10'
      autoid={autoid ? generateAutomationId(`${autoid}_checkbox`) : 'checkbox'}
      data-checked={checked}
      data-indeterminate={indeterminate}
      data-disabled={disabled}
      onClick={handleOnClick}
      {...(disabled ? disabledStyles : {})}
      {...(checked ? checkedStyles : {})}
    >
      {(indeterminate || checked) && (
        <Icon
          autoid={`${autoid && `${autoid}_`}checkbox_icon`}
          name={indeterminate ? 'minus' : 'check'}
          noFill
          size='full'
        />
      )}
    </Interactable>
  );
};

export default Checkbox;
