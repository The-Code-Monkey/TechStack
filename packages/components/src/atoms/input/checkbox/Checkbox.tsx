import { Interactable, Icon } from '../../../primal';
import { generateAutomationId } from '../../../utils';

import { disabledStyles, checkedStyles } from './styled';

export interface Props {
  testid?: string;
  onClick?: (p1: boolean, p2: MouseEvent) => void;
  checked?: boolean;
  indeterminate?: boolean;
  disabled?: boolean;
  name?: string;
}

const Checkbox = ({
  indeterminate,
  checked,
  onClick,
  disabled,
  testid,
  name,
}: Props) => {
  const handleOnClick = (event: MouseEvent) => {
    const status = indeterminate || !checked;

    if (onClick) {
      onClick(status, event);
    }
  };

  return (
    <Interactable
      name={name}
      display='flex'
      size={7}
      alignItems='center'
      justifyContent='center'
      bgColor='neutrals.0'
      border='1'
      borderRadius='1'
      borderColor='neutrals.10'
      testid={testid ? generateAutomationId(`${testid}_checkbox`) : 'checkbox'}
      data-checked={checked}
      data-indeterminate={indeterminate}
      data-disabled={disabled}
      onClick={handleOnClick}
      {...(disabled ? disabledStyles : {})}
      {...(checked ? checkedStyles : {})}
    >
      {(indeterminate || checked) && (
        <Icon
          testid={
            testid
              ? `${testid}_checkbox_icon`
              : indeterminate
              ? 'minus'
              : 'check'
          }
          name={indeterminate ? 'minus' : 'check'}
          size='full'
        />
      )}
    </Interactable>
  );
};

export default Checkbox;
