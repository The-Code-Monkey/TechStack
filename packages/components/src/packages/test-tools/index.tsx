import { mount, shallow, ShallowWrapper, ReactWrapper } from 'enzyme';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: any) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some((prop) => keys.some((key) => key === prop)),
    print: (val: any, serialize: any) => {
      keys.forEach((key) => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const shallowWithTheme: (children: ReactNode) => ShallowWrapper = (
  children: ReactNode
) =>
  shallow(<ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>)
    .dive()
    .shallow();

export const mountWithTheme: (children: ReactNode) => ReactWrapper = (
  children: ReactNode
) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
