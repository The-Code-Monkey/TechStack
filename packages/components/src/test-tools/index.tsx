import isPropValid from '@emotion/is-prop-valid';
import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { StyleSheetManager, ThemeProvider } from 'styled-components';
import { expect } from 'bun:test';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode): RenderResult => {
  // expect.addSnapshotSerializer(removeProperties());

  return render(
    <StyleSheetManager shouldForwardProp={isPropValid}>
      <ConfigContext.Provider value={config}>
        <ThemeProvider theme={formattedTheme}>
          <>{children}</>
        </ThemeProvider>
      </ConfigContext.Provider>
    </StyleSheetManager>
  );
};
