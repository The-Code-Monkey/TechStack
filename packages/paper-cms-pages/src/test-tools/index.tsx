import { ConfigContext } from '@techstack/components';
import { render } from '@testing-library/react';
import { ReactNode } from 'react';

import theme from '../theme';
import { ThemeProvider } from '../workarounds/styled-components';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode) => {
  expect.addSnapshotSerializer(removeProperties());

  return render(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
