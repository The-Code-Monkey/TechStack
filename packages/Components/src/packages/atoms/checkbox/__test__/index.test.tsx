import React from 'react';
import toJson from 'enzyme-to-json';

import { mountWithTheme } from '../../../test-tools';
import Checkbox from '..';

describe('<Checkbox />', () => {
  it('renders correctly no icon', () => {
    const wrapper = mountWithTheme(<Checkbox />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly checked', () => {
    const wrapper = mountWithTheme(<Checkbox checked />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly indeterminate', () => {
    const wrapper = mountWithTheme(<Checkbox indeterminate />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls onClick function', () => {
    const handleClick = jest.fn();

    const wrapper = mountWithTheme(<Checkbox onClick={handleClick} />);

    wrapper.find('div').simulate('click');

    expect(handleClick).toHaveBeenCalled();
  });
});
