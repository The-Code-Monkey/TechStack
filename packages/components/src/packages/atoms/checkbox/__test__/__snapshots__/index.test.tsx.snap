// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly checked 1`] = `
<Component>
  <Checkbox
    checked={true}
  >
    <Interactable
      alignItems="center"
      autoid="checkbox"
      bgColor="neutrals.0"
      border="1"
      borderColor="neutrals.10"
      borderRadius="1"
      cursor="pointer"
      data-checked={true}
      display="flex"
      justifyContent="center"
      onClick={[Function]}
      opacity="3"
      size={7}
    >
      <Styled(Box)
        alignItems="center"
        bgColor="neutrals.0"
        border="1"
        borderColor="neutrals.10"
        borderRadius="1"
        cursor="pointer"
        data-autoid="checkbox"
        data-checked={true}
        display="flex"
        justifyContent="center"
        onClick={[Function]}
        opacity="3"
        size={7}
      >
        <Box
          alignItems="center"
          bgColor="neutrals.0"
          border="1"
          borderColor="neutrals.10"
          borderRadius="1"
          className="sc-hKMtZM"
          cursor="pointer"
          data-autoid="checkbox"
          data-checked={true}
          display="flex"
          justifyContent="center"
          onClick={[Function]}
          opacity="3"
          size={7}
        >
          <styled.div
            alignItems="center"
            bgColor="neutrals.0"
            border="1"
            borderColor="neutrals.10"
            borderRadius="1"
            className="sc-hKMtZM"
            cursor="pointer"
            data-autoid="checkbox"
            data-checked={true}
            display="flex"
            justifyContent="center"
            onClick={[Function]}
            opacity="3"
            size={7}
          >
            <div
              className="sc-gsnTZi knjsmK sc-hKMtZM"
              cursor="pointer"
              data-autoid="checkbox"
              data-checked={true}
              onClick={[Function]}
            >
              <Memo(Icon)
                autoid="undefinedcheckbox_icon"
                name="Check"
                noFill={true}
                size="full"
              >
                <Suspense
                  fallback={<ForwardRef(HelpCircle) />}
                >
                  <Styled(Box)
                    autoid="undefinedcheckbox_icon_icon"
                    noFill={true}
                    size="full"
                  >
                    <Box
                      autoid="undefinedcheckbox_icon_icon"
                      className="sc-dkzDqf Vwowz"
                    >
                      <styled.div
                        className="sc-dkzDqf Vwowz"
                        data-autoid="undefinedcheckbox_icon_icon"
                      >
                        <div
                          className="sc-gsnTZi cXXFdo sc-dkzDqf Vwowz"
                          data-autoid="undefinedcheckbox_icon_icon"
                        />
                      </styled.div>
                    </Box>
                  </Styled(Box)>
                  <ForwardRef(HelpCircle)>
                    <svg
                      fill="none"
                      height={24}
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                      <path
                        d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                      />
                      <line
                        x1="12"
                        x2="12.01"
                        y1="17"
                        y2="17"
                      />
                    </svg>
                  </ForwardRef(HelpCircle)>
                </Suspense>
              </Memo(Icon)>
            </div>
          </styled.div>
        </Box>
      </Styled(Box)>
    </Interactable>
  </Checkbox>
</Component>
`;

exports[`<Checkbox /> renders correctly indeterminate 1`] = `
<Component>
  <Checkbox
    indeterminate={true}
  >
    <Interactable
      alignItems="center"
      autoid="checkbox"
      bgColor="neutrals.0"
      border="1"
      borderColor="neutrals.10"
      borderRadius="1"
      cursor="pointer"
      data-indeterminate={true}
      display="flex"
      justifyContent="center"
      onClick={[Function]}
      size={7}
    >
      <Styled(Box)
        alignItems="center"
        bgColor="neutrals.0"
        border="1"
        borderColor="neutrals.10"
        borderRadius="1"
        cursor="pointer"
        data-autoid="checkbox"
        data-indeterminate={true}
        display="flex"
        justifyContent="center"
        onClick={[Function]}
        size={7}
      >
        <Box
          alignItems="center"
          bgColor="neutrals.0"
          border="1"
          borderColor="neutrals.10"
          borderRadius="1"
          className="sc-hKMtZM"
          cursor="pointer"
          data-autoid="checkbox"
          data-indeterminate={true}
          display="flex"
          justifyContent="center"
          onClick={[Function]}
          size={7}
        >
          <styled.div
            alignItems="center"
            bgColor="neutrals.0"
            border="1"
            borderColor="neutrals.10"
            borderRadius="1"
            className="sc-hKMtZM"
            cursor="pointer"
            data-autoid="checkbox"
            data-indeterminate={true}
            display="flex"
            justifyContent="center"
            onClick={[Function]}
            size={7}
          >
            <div
              className="sc-gsnTZi lkeFYh sc-hKMtZM"
              cursor="pointer"
              data-autoid="checkbox"
              data-indeterminate={true}
              onClick={[Function]}
            >
              <Memo(Icon)
                autoid="undefinedcheckbox_icon"
                name="Minus"
                noFill={true}
                size="full"
              >
                <Suspense
                  fallback={<ForwardRef(HelpCircle) />}
                >
                  <Styled(Box)
                    autoid="undefinedcheckbox_icon_icon"
                    noFill={true}
                    size="full"
                  >
                    <Box
                      autoid="undefinedcheckbox_icon_icon"
                      className="sc-dkzDqf Vwowz"
                    >
                      <styled.div
                        className="sc-dkzDqf Vwowz"
                        data-autoid="undefinedcheckbox_icon_icon"
                      >
                        <div
                          className="sc-gsnTZi cXXFdo sc-dkzDqf Vwowz"
                          data-autoid="undefinedcheckbox_icon_icon"
                        />
                      </styled.div>
                    </Box>
                  </Styled(Box)>
                  <ForwardRef(HelpCircle)>
                    <svg
                      fill="none"
                      height={24}
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                      <path
                        d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                      />
                      <line
                        x1="12"
                        x2="12.01"
                        y1="17"
                        y2="17"
                      />
                    </svg>
                  </ForwardRef(HelpCircle)>
                </Suspense>
              </Memo(Icon)>
            </div>
          </styled.div>
        </Box>
      </Styled(Box)>
    </Interactable>
  </Checkbox>
</Component>
`;

exports[`<Checkbox /> renders correctly no icon 1`] = `
<Component>
  <Checkbox>
    <Interactable
      alignItems="center"
      autoid="checkbox"
      bgColor="neutrals.0"
      border="1"
      borderColor="neutrals.10"
      borderRadius="1"
      cursor="pointer"
      display="flex"
      justifyContent="center"
      onClick={[Function]}
      size={7}
    >
      <Styled(Box)
        alignItems="center"
        bgColor="neutrals.0"
        border="1"
        borderColor="neutrals.10"
        borderRadius="1"
        cursor="pointer"
        data-autoid="checkbox"
        display="flex"
        justifyContent="center"
        onClick={[Function]}
        size={7}
      >
        <Box
          alignItems="center"
          bgColor="neutrals.0"
          border="1"
          borderColor="neutrals.10"
          borderRadius="1"
          className="sc-hKMtZM"
          cursor="pointer"
          data-autoid="checkbox"
          display="flex"
          justifyContent="center"
          onClick={[Function]}
          size={7}
        >
          <styled.div
            alignItems="center"
            bgColor="neutrals.0"
            border="1"
            borderColor="neutrals.10"
            borderRadius="1"
            className="sc-hKMtZM"
            cursor="pointer"
            data-autoid="checkbox"
            display="flex"
            justifyContent="center"
            onClick={[Function]}
            size={7}
          >
            <div
              className="sc-gsnTZi lkeFYh sc-hKMtZM"
              cursor="pointer"
              data-autoid="checkbox"
              onClick={[Function]}
            />
          </styled.div>
        </Box>
      </Styled(Box)>
    </Interactable>
  </Checkbox>
</Component>
`;
